Add colorInteropID standard attribute

cary-ilm opened the pull request:
See [https://lf-aswf.atlassian.net/wiki/spaces/CIF/overview](https://lf-aswf.atlassian.net/wiki/spaces/CIF/overview) for discussion

TODO: fill in the proper documentation

<!-- readthedocs-preview openexr start -->
Website preview: https://openexr--2075.org.readthedocs.build/en/2075/
<!-- readthedocs-preview openexr end -->

cary-ilm committed:
Add colorInteropID standard attribute

See https://lf-aswf.atlassian.net/wiki/spaces/CIF/overview for discussion

TODO: fill in the proper documentation

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented:
@carolalynn, @doug-walker, and @peterhillman, this should be the basic structure of the color interop attribute, please help confirm the details, i.e. exact name, code comments, and website documentation.

doug-walker commented on website/TechnicalIntroduction.rst:
```diff
@@ -1313,6 +1313,9 @@ By default, OpenEXR files have the following attributes:
   <https://dl.acm.org/doi/abs/10.1145/3233085.3233086>`_ for details.
 
+**colorInteropID**
+  Color Interop ID. See `Color Interop Forum <https://lf-aswf.atlassian.net/wiki/spaces/CIF/overview>`_
```
```suggestion
  Color Interop ID. Provides a mechanism to identify the color space of the RGB images. 
  See `An ID for Color Interop <https://docs.google.com/document/d/1T94lYbis9uCskL_ZEMxGBF2JryLfZnjxlEoNgRHZzBE/edit?usp=sharing>`_ 
  and `Identifying the Color Space of OpenEXR Files <https://docs.google.com/document/d/1MTH1bq2L67ifvdDf64Amhzg4AbkIM5LG6yPHrB96Vwo/edit?usp=sharing>`_ for details.
```

doug-walker commented on website/StandardAttributes.rst:
```diff
@@ -1376,6 +1376,23 @@ pipeline. All are optional.
+       </p>
+       
+       <p style="padding-bottom:15px">
+       Add description here
```
```suggestion
         See <a href=https://docs.google.com/document/d/1T94lYbis9uCskL_ZEMxGBF2JryLfZnjxlEoNgRHZzBE/edit?usp=sharing>An ID for Color Interop</a> 
         and <a href=https://docs.google.com/document/d/1MTH1bq2L67ifvdDf64Amhzg4AbkIM5LG6yPHrB96Vwo/edit?usp=sharing>Identifying the Color Space of OpenEXR Files</a> for details.
```

doug-walker commented on website/StandardAttributes.rst:
```diff
@@ -1376,6 +1376,23 @@ pipeline. All are optional.
+     </tt> </td>
+     <td style="vertical-align: top; width:500px">
+       <p style="padding-bottom:15px">
+         Color Interop ID
```
```suggestion
         Provides a mechanism to identify the color space of the RGB images.
```

doug-walker commented on src/test/OpenEXRTest/testStandardAttributes.cpp:
```diff
@@ -906,6 +906,54 @@ generatedFunctions ()
+    Header header (W, H);
+    assert (hasColorInteropID (header) == false);
+
+    std::string id1 = "rec2100_pq_display";
```
```suggestion
    std::string id1 = "lin_ap1_scene";
```

doug-walker commented on src/lib/OpenEXR/ImfStandardAttributes.h:
```diff
@@ -724,4 +724,12 @@ IMF_STD_ATTRIBUTE_DEF_DEPRECATED (
 IMF_STD_ATTRIBUTE_DEF (idManifest, IDManifest, CompressedIDManifest)
 
+//
+// ColorInteropID - the OpenColorIO color space/config name
```
```suggestion
// ColorInteropID -- the Color Interop Forum ID that describes the color space of the RGB images.
```

doug-walker commented on src/lib/OpenEXR/ImfStandardAttributes.h:
```diff
@@ -724,4 +724,12 @@ IMF_STD_ATTRIBUTE_DEF_DEPRECATED (
+//
+// ColorInteropID - the OpenColorIO color space/config name
+//
+// [add description here]
```
```suggestion
// This is a specially designed string used to communicate the color space of 
// the image in an interoperable manner. The contents of the string is described
// in the specification "An ID for Color Interop":
// <https://docs.google.com/document/d/1T94lYbis9uCskL_ZEMxGBF2JryLfZnjxlEoNgRHZzBE/edit?usp=sharing>
//
// Guidance to application developers is provided in "Identifying the Color Space of OpenEXR Files":
// <https://docs.google.com/document/d/1MTH1bq2L67ifvdDf64Amhzg4AbkIM5LG6yPHrB96Vwo/edit?usp=sharing>
// 
// Please adhere to the following usage guidance:
//   1. If the color space of an image is modified, do not forward the stale attribute value.
//   2. Do not set this to a default color space. If the actual color space is not known, omit
//      the attribute or set it to "unknown".
```

doug-walker reviewed (commented):
Thank you Cary!  Here's how I would fill in the various descriptions.

I'll do a follow-up PR after the recommendations have been moved to the ColorInterop GitHub repo, as MarkDown files.

cary-ilm committed:
Update website/TechnicalIntroduction.rst

Co-authored-by: Doug Walker <doug.walker@autodesk.com>
Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Update website/StandardAttributes.rst

Co-authored-by: Doug Walker <doug.walker@autodesk.com>
Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Update website/StandardAttributes.rst

Co-authored-by: Doug Walker <doug.walker@autodesk.com>
Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Update src/lib/OpenEXR/ImfStandardAttributes.h

Co-authored-by: Doug Walker <doug.walker@autodesk.com>
Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Update src/test/OpenEXRTest/testStandardAttributes.cpp

Co-authored-by: Doug Walker <doug.walker@autodesk.com>
Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Update src/lib/OpenEXR/ImfStandardAttributes.h

Co-authored-by: Doug Walker <doug.walker@autodesk.com>
Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Add note that colorInteropID is new in v3.4

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented:
Thanks, Doug, that looks good to me.  I added a note that colorInteropID is new in v3.4.  The website will go live with this change when it's merged, so I added a "not yet released" note as well, which I'll remove once the release is out.

cary-ilm commented:
Merging this as is for now, we can follow up with further refinements as needed.
